packages/cmf/src/onError.js

```diff
@@ -131,24 +181,38 @@ function getErrors() {
 	return ref.errors;
 }
 
-/**
- * @return {Boolean} true if we can do report to backend
- */
-function hasReportURL() {
-	return !!ref.serverURL;
+function setupFromSettings(settings) {
+	const dsn = get(settings, 'env.SENTRY_DSN');
+	if (ref.SENTRY_DSN !== dsn && !ref.SENTRY_DSN) {
+		ref.SENTRY_DSN = dsn;
+		setupSentry();
+	}
 }
 
 /**
- * simple try catch middleware for redux
+ * onError redux middleware.
+ * it store last 20 actions
+ * it catch settings fetch OK to try to setup Sentry
+ * it try catch every sub actions effect to report error
  */
 function middleware() {
 	return next => action => {
+		if (!ref.SENTRY_DSN) {
+			if (ref.actions.length >= 20) {
+				ref.actions.shift();
+			}
+			ref.actions.push(action && action.type ? action.type : 'UNKNOWN');
```

Comment:
might push `undefined`
```suggestion
			ref.actions.push(get(action, 'type', 'UNKNOWN'));
```

Reply:
no the original code is working but ok for the change